### PR TITLE
feat(behavior_path_planner): all modules publish modified goal

### PR DESCRIPTION
**Pull request by kosuke55**
Signed-off-by: kosuke55 <kosuke.tnp@gmail.com>

## Description

all modules of behaivor_path publish modified_goal.

## Related links

related mission planner update: https://github.com/autowarefoundation/autoware.universe/pull/1495
> // TODO(Takagi, Isamu): remove when modified goal is always published

launcher:  https://github.com/tier4/autoware_launch/pull/519

[TIERIV INTERNAL LINK](https://tier4.atlassian.net/browse/T4PB-22283)

## Tests performed

psim

check in all modules. 
(~~wip: currently use current stamp and publish goal pose at 10hz, think about which stamp is better~~
-> route stamp is better to detect new route, so changed to it) 

https://user-images.githubusercontent.com/39142679/196347912-1ab78bc9-38c1-4fb0-a258-733014942c0a.mp4

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp [hunk 1]
```diff
@@ -2052,6 +2052,10 @@ BehaviorModuleOutput AvoidanceModule::plan()
   }
   output.path = std::make_shared<PathWithLaneId>(avoidance_path.path);
 
+  output.modified_goal.header.frame_id = planner_data_->route_handler->getRouteHeader().frame_id;
+  output.modified_goal.header.stamp = clock_->now();
```
Comment (kosuke55) [inline planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp:2056]: think about which stamp is better.

Comment (kosuke55) [reply planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp]: chaned to route stamp in 
https://github.com/autowarefoundation/autoware.universe/pull/2094/commits/809a125f05665f7c37171cb206ca22a00b505ca0

Comment (kosuke55) [reply planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp]: tier4 interanal slack https://star4.slack.com/archives/C4P0NSMB5/p1666069602035659?thread_ts=1665625570.381429&cid=C4P0NSMB5

File: planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp [hunk 2]
```diff
@@ -2119,6 +2122,8 @@ BehaviorModuleOutput AvoidanceModule::planWaitingApproval()
 {
   // we can execute the plan() since it handles the approval appropriately.
   BehaviorModuleOutput out = plan();
+  out.modified_goal.header.frame_id = planner_data_->route_handler->getRouteHeader().frame_id;
+  out.modified_goal.header.stamp = clock_->now();
```
Comment (yukkysaito) [inline planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp:2126]: stamp should be used the trigger time?

**Comment by kosuke55** [comment]
after discussing in https://github.com/orgs/autowarefoundation/discussions/2983, we changed the way.
closed this PR, and send new PR https://github.com/autowarefoundation/autoware.universe/pull/2602

**Pull request closed without merging**